rpc2/transport.go

```diff
@@ -204,27 +210,25 @@ func (t *Transport) ReadByte() (b byte, err error) {
 }
 
 func (t *Transport) Decode(i interface{}) (err error) {
-	var cp *ConPackage
+	var cp *conPackage
 	if cp, err = t.getConPackage(); err == nil {
 		err = cp.Decode(i)
 	}
-	return
+	return err
 }
 
 func (t *Transport) RawWrite(b []byte) (err error) {
-	var cp *ConPackage
+	var cp *conPackage
 	if cp, err = t.getConPackage(); err == nil {
 		err = cp.Write(b)
 	}
 	return err
 }
 
-func (t *Transport) GetDispatcher() (d Dispatcher, err error) {
-	t.run(true)
+func (t *Transport) getDispatcher() (dispatcher, error) {
 	if !t.IsConnected() {
-		err = DisconnectedError{}
+		return nil, DisconnectedError{}
```

Comment:
@akalin-keybase brings up a good point, lintable code is also good wherever possible.  Again, not something I was wise enough to do at first.